test/project.e2e-spec.ts

```diff
@@ -1140,4 +1140,42 @@ describe('Project e2e', () => {
       projectQueryResult.project.budget.value.records[0].organization.value.id;
     expect(firstBudgetRecordOrganizationId).toBe(org.id);
   });
+
+  it('should not assign the same department id to two projects created with the same location', async () => {
+    await loginAsAdmin(app);
```

Comment:
should it not be used anywhere?

Reply:
Also speaking of problems with concurrent requests. I tried `Promise.all`-ing the creation and updating of two projects and the ended up with the same dept id. If done synchronously, as is done in the test now, they don't.

Reply:
Adding the label with the unique constraint would cause this to fail instead of creating a duplicate. Maybe start with that to confirm the problem.

Reply:
That's what I was hoping. It seems like we should lock all unique properties if we're doing that, not just this one.

Reply:
Neo4j normally handles the locking internally, and unique constraint violations just throw an error so there's no reason to do anything different normally. This is an edge case where we have complex query logic that neo4j can't figure out implicitly.

Reply:
When do constraints get applied?

Reply:
So the constraint is working if I try and change a dept id in the db to the same as another. It isn't working in the test suite. I'm wondering if the constraint isn't getting applied.

Reply:
Constraints are applied when app starts & db connects. Unless you have
```
DB_CREATE_INDEXES=false
```
in your .env.local

Did you add the label [here](https://github.com/SeedCompany/cord-api-v3/pull/1731/files#diff-d4b319fbc9fb22707d5969d0591543ce306f553ffcadfd4397fc912029545b4eR58)?

Reply:
So the constraint works, but not for tests. Dunno if it's not getting applied properly. 

You're saying that in this case it's not sufficient to have a unique constraint? Will it leave the db in a partial state?

Reply:
The resetDatabase call removes all constraints after the test suite, maybe that's some how applying out of order. I wouldn't worry about it too much because 👇 

> You're saying that in this case it's not sufficient to have a unique constraint? Will it leave the db in a partial state?

No it's not sufficient here because the change isn't directly from a user. They can't help if the same dept id was chosen at the same time as another.

Also yes, almost all errors will leave database in a partial state until transactions can be applied.

Reply:
I did add the label. I should push my changes so far.